exchanges/huobi/huobi_test.go

```diff
@@ -3003,3 +2015,20 @@ func TestBootstrap(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, h.futureContractCodes)
 }
+
+var updatePairsMutex sync.Mutex
+
+func updatePairsOnce(tb testing.TB) {
+	tb.Helper()
+
+	updatePairsMutex.Lock()
+	defer updatePairsMutex.Unlock()
+
+	testexch.UpdatePairsOnce(tb, h)
+
+	p, err := h.pairFromContractExpiryCode(btccwPair)
+	require.NoError(tb, err, "pairFromContractCode must not error")
+	err = h.CurrencyPairs.EnablePair(asset.Futures, p)
+	require.NoError(tb, common.ExcludeError(err, currency.ErrPairAlreadyEnabled))
+	btcFutureDatedPair = p
```

Comment:
[RACE](https://github.com/thrasher-corp/gocryptotrader/actions/runs/12289907293/job/34296163066?pr=1703#step:10:765) 🚗 

Maybe have a global bool above with the mutex so we don't set this value again

Reply:
We have a bool type already; if btcFutureDatedPair == currency.EMPTYPAIR.
However for some reason that's not enough.

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/d176b4911

This turned out to be relatively complicated; I forgot that the shared test utility `updatePairsOnce` will nuke the `CurrencyPairs`. That means that not only was it unsafe to call the local `updatePairsOnce` in huobi_test without re-enabling the (already found) pair, but also anything else that Enables a pair should localise it's exchange instance `h` to avoid getting clobbered. In this case `TestGetLatestFundingRates`